cmd/claimserver/endpoint/admin.go

```diff
@@ -0,0 +1,66 @@
+package endpoint
+
+import (
+	"net/http"
+	"strconv"
+
+	"github.com/gin-gonic/gin"
+)
+
+func handleInfo(c *gin.Context) {
+	r := adminservice.Info()
+	c.JSON(200, gin.H{
+		"info": r,
+	})
+}
+
+func handleRawDump(c *gin.Context) {
+	r := adminservice.RawDump()
+	c.JSON(http.StatusOK, r)
+}
+
+func handleRawImport(c *gin.Context) {
+	var data map[string]string
+	c.BindJSON(&data)
+
+	count, err := adminservice.RawImport(data)
+	if err != nil {
+		c.String(http.StatusBadRequest, err.Error())
+	}
+	c.String(http.StatusOK, "imported "+strconv.Itoa(count)+" key,value entries")
+}
+
+func handleClaimsDump(c *gin.Context) {
+	r := adminservice.ClaimsDump()
+	c.JSON(http.StatusOK, r)
+}
+
+type addClaimBasicMsg struct {
+	Namespace string
+	IndexData string
+	Data      string
+}
+
+func handleAddClaimBasic(c *gin.Context) {
+	var m addClaimBasicMsg
+	c.BindJSON(&m)
+
+	if len(m.IndexData) != 400/8 {
+		c.String(http.StatusBadRequest, "indexData smaller than 400/8")
+		return
+	}
+	if len(m.Data) != 496/8 {
+		c.String(http.StatusBadRequest, "data smaller than 496/8")
+		return
+	}
+
+	var indexSlot [400 / 8]byte
+	var dataSlot [496 / 8]byte
+	copy(indexSlot[:], m.IndexData[:400/8])
```

Comment:
You are copying a string into a binary array.  Is this what you wanted?

Reply:
solved